homeassistant/components/media_player/plex.py

```diff
@@ -272,93 +527,383 @@ def _active_media_plexapi_type(self):
+                    return self._convert_na_to_none(self.session.parentTitle)
+
+    @property
+    def media_album_artist(self):
+        """Album artist of current playing media, music track only."""
+        if self.state not in (STATE_IDLE, STATE_OFF):  # true or frozen state
+            if self.media_content_type is MEDIA_TYPE_MUSIC:
+                if self.session:
+                    return self._convert_na_to_none(
+                        self.session.grandparentTitle)
+
+    @property
+    def media_track(self):
+        """Track number of current playing media, music track only."""
+        if self.state not in (STATE_IDLE, STATE_OFF):  # true or frozen state
+            if self.media_content_type is MEDIA_TYPE_MUSIC:
+                if self.session:
+                    return self._convert_na_to_none(self.session.index)
 
     @property
     def media_duration(self):
         """Duration of current playing media in seconds."""
-        if self.session is not None:
-            return self._convert_na_to_none(self.session.duration)
+        if self.state not in (STATE_IDLE, STATE_OFF):  # true or frozen state
+            if self.session:
+                return self._convert_na_to_none(self.session.duration)
+
+    @property
+    def media_position(self):
+        """Position of current playing media in seconds."""
+        if self.session:
+            return self._convert_na_to_none(self.session.viewOffset)
+
+    @property
+    def media_position_updated_at(self):
+        """When was the position of the current playing media valid."""
+        if self.state not in (STATE_IDLE, STATE_OFF):  # true or frozen state
+            if self.state == STATE_PLAYING:
+                self._media_position_updated_at = util.dt.utcnow()
+            return self._media_position_updated_at
 
     @property
     def media_image_url(self):
         """Image url of current playing media."""
-        if self.session is not None:
-            thumb_url = self._convert_na_to_none(self.session.thumbUrl)
-            if str(self.na_type) in thumb_url:
-                # Audio tracks build their thumb urls internally before passing
-                # back a URL with the PlexAPI _NA type already converted to a
-                # string and embedded into a malformed URL
-                thumb_url = None
-            return thumb_url
+        if self.state not in (STATE_IDLE, STATE_OFF):  # true or frozen state
+            if self.session:
+                thumb_url = self._convert_na_to_none(self.session.thumb)
+                if self.media_content_type is MEDIA_TYPE_TVSHOW:
+                    if not self.optional_config[CONF_USE_EPISODE_ART]:
+                        thumb_url = self._convert_na_to_none(
+                            self.session.grandparentThumb)
+
+                if thumb_url:
+                    thumb_url = self.session.server.url(thumb_url)
+                    thumb_response = requests.get(thumb_url, verify=False)
```

Comment:
If we need to do web requests to validate a property, this should happen inside the `update` method.